### PR TITLE
Generate GtkGestureStylus

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
In the continuation of my previous merge requests. 

I was wondering about two enums which are deprecated in 3.20, should be generated? 
- `Gtk.ArrowPlacement`
- `Gtk.ToolbarSpaceStyle`

### CODE REVIEW

**Comment by sdroege** [comment]
> I was wondering about two enums which are deprecated in 3.20, should be generated?
> 
>     * `Gtk.ArrowPlacement`
> 
>     * `Gtk.ToolbarSpaceStyle`

I think they're only used in GObject properties. Would we get those if we generate them, or were those properties deprecated before 3.14? I'd add them for now if that causes new API to be generated.

File: src/auto/accel_map.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    //}
+
+    pub fn lock_path(accel_path: &str) {
+        assert_initialized_main_thread!();
+        unsafe {
+            gtk_sys::gtk_accel_map_lock_path(accel_path.to_glib_none().0);
+        }
+    }
+
+    //pub fn lookup_entry(accel_path: &str, key: /*Ignored*/AccelKey) -> bool {
```
Comment (sdroege) [inline src/auto/accel_map.rs:102]: Is this class useful without this function?

File: src/auto/enums.rs [hunk 2]
```diff
@@ -1729,6 +1729,86 @@ impl SetValue for EventSequenceState {
...
+pub enum ExpanderStyle {
+    Collapsed,
+    SemiCollapsed,
+    SemiExpanded,
+    Expanded,
+    #[doc(hidden)]
+    __Unknown(i32),
+}
+
+impl fmt::Display for ExpanderStyle {
```
Comment (sdroege) [inline src/auto/enums.rs:1743]: This does not seem to be used in any API we have, so just having the enum generated wouldn't provide anything useful.

File: src/auto/enums.rs [hunk 3]
```diff
@@ -6512,6 +6592,84 @@ impl SetValue for StateType {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Clone, Copy)]
+#[non_exhaustive]
+pub enum TextBufferTargetInfo {
```
Comment (sdroege) [inline src/auto/enums.rs:6597]: This does not seem to be used in any API we have, so just having the enum generated wouldn't provide anything useful.

File: src/auto/flags.rs [hunk 4]
```diff
@@ -111,6 +111,55 @@ impl SetValue for ApplicationInhibitFlags {
     }
 }
 
+bitflags! {
```
Comment (sdroege) [inline src/auto/flags.rs:114]: Also this doesn't seem used anywhere

File: src/auto/flags.rs [hunk 5]
```diff
@@ -215,6 +264,74 @@ impl SetValue for CellRendererState {
...
+    pub struct DebugFlag: u32 {
```
Comment (sdroege) [inline src/auto/flags.rs:268]: And not this

File: src/auto/gradient.rs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    match fn {
+        copy => |ptr| gobject_sys::g_boxed_copy(gtk_sys::gtk_gradient_get_type(), ptr as *mut _) as *mut gtk_sys::GtkGradient,
+        free => |ptr| gobject_sys::g_boxed_free(gtk_sys::gtk_gradient_get_type(), ptr as *mut _),
+        get_type => || gtk_sys::gtk_gradient_get_type(),
+    }
+}
+
+impl Gradient {
+    pub fn resolve_for_context<P: IsA<StyleContext>>(&self, context: &P) -> Option<cairo::Pattern> {
```
Comment (sdroege) [inline src/auto/gradient.rs:23]: The whole type is deprecated since 3.8, just in this function it was forgotten to specify the version

**Comment by bilelmoussaoui** [comment]
I should have waited till morning before submitting this, removed the unused enums.

I have only kept Gtk.GestureStylus & Gtk.DebugFlag

**Comment by bilelmoussaoui** [comment]
> > I was wondering about two enums which are deprecated in 3.20, should be generated?
> > ```
> > * `Gtk.ArrowPlacement`
> > 
> > * `Gtk.ToolbarSpaceStyle`
> > ```
> 
> I think they're only used in GObject properties. Would we get those if we generate them, or were those properties deprecated before 3.14? I'd add them for now if that causes new API to be generated.

they don't generate any new API

**Comment by sdroege** [comment]
> I have only kept [...] Gtk.DebugFlag

That one also doesn't seem to be used anywhere.

File: src/auto/gesture_stylus.rs [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        skip_assert_initialized!();
+        unsafe {
+            Gesture::from_glib_full(gtk_sys::gtk_gesture_stylus_new(
+                widget.as_ref().to_glib_none().0,
+            ))
+            .unsafe_cast()
+        }
+    }
+
+    //#[cfg(any(feature = "v3_24", feature = "dox"))]
```
Comment (bilelmoussaoui) [inline src/auto/gesture_stylus.rs:49]: @sdroege any idea how can I generate this one? :)

Comment (sdroege) [reply src/auto/gesture_stylus.rs:49]: You have to do it manually, see the docs of this function in C: `axes: array of requested axes, terminated with GDK_AXIS_IGNORE`

**Comment by bilelmoussaoui** [comment]
Updated with `get_axes` manual binding as well

**Comment by bilelmoussaoui** [comment]
Somehow the CI failed to report it's state, but everything is green :)

**Pull request merged**